show_graph.py

```diff
@@ -21,44 +18,62 @@ def DrawBarGraph(connection, table_name, y_axis_field, x_axis_values,
-    if not direct_list:
-      print "{} {} is not found in table for direct results.".format(
-          x_axis_field, x)
-      direct_list = [0]
-
-    direct_lists.append(direct_list)
-
-    condition = ("where {}=\"{}\" and"
-                 " category=\"envoy\"").format(x_axis_field, x)
-    envoy_list = db_utils.SingleColumnToList(db_utils.GetFieldFromTable(
-        connection, table_name, field=y_axis_field, cond=condition))
-    if not envoy_list:
-      print "{} {} is not found in table for Envoy results.".format(
-          x_axis_field, x)
-      envoy_list = [0]
-
-    envoy_lists.append(envoy_list)
+
+  def GetListsFromDB(x_axis_values, x_axis_field, connection,
+                     table_name, y_axis_field, category):
+    """This function returns lists of values of a field from the DB.
+
+    The function returns lists of `y_axis_field` for the values corresponding to
+    the `x_axis_values` in `x_axis_field`.
+    Args:
+      x_axis_values: a list of values for which the `y_axis_field` will be
+       fetched for.
+      x_axis_field: name of the field for x_axis
+      connection: the connection to the database
+      table_name: name of the table in the database which has the data
+      y_axis_field: the name of the column in the table, whose data will be put
+      into the list
+      category: Direct or Envoy or which category the data belong to
+    Returns:
+      Returns a list of lists with all the values of `y_axis_field`
+      corresponding to `x_axis_values`.
+    """
+    lists = list()
+    for x in x_axis_values:
+      condition = ("where {}=\"{}\" and"
+                   " category=\"{}\"").format(x_axis_field, x, category)
+      single_list = db_utils.SingleColumnToList(db_utils.GetFieldFromTable(
+          connection, table_name, field=y_axis_field, cond=condition))
+      if not single_list:
+        print "{} {} is not found in table for {} results.".format(
+            x_axis_field, x, category)
+        single_list = [0]
+
+      lists.append(single_list)
+    return lists
+
+  direct_lists = GetListsFromDB(x_axis_values, x_axis_field, connection,
+                                table_name, y_axis_field, "direct")
+  envoy_lists = GetListsFromDB(x_axis_values, x_axis_field, connection,
+                               table_name, y_axis_field, "envoy")
 
   ind = np.arange(len(x_axis_values))
-  direct_means = list()
-  direct_std = list()
-  envoy_means = list()
-  envoy_std = list()
-
-  for single_set in direct_lists:
-    direct_means.append(np.mean(single_set))
-    direct_std.append(np.std(single_set))
-  for single_set in envoy_lists:
-    envoy_means.append(np.mean(single_set))
-    envoy_std.append(np.std(single_set))
+
+  def GetMeansAndStdsFromList(lists):
+    """This function returns the means and standard deviation of lists.
+
+    Args:
+      lists: A list of lists. Each list inside the top-level list consists a
+      bunch of numbers
```

Comment:
".. consists of a sample for a given variable that summary stats will be computed on."